### PR TITLE
Correction of DroneConnect

### PR DESCRIPTION
**Pull request by Anghul**
Correction of all bugs found when connecting a drone, and addition of an animation while connecting to a drone to show the user something is happening

### CODE REVIEW

File: app/src/androidTest/java/ch/epfl/sdp/drone3d/drone/DroneServiceTest.kt [hunk 1]
```diff
@@ -45,6 +52,8 @@ class DroneServiceTest {
 
         verify(factory).createSimulation(ip, port)
 
+        // now stop isConnected to have to return true
+        `when`(drone.core.connectionState).thenCallRealMethod()
```
Comment (GabrielFleischer) [inline app/src/androidTest/java/ch/epfl/sdp/drone3d/drone/DroneServiceTest.kt:56]: I am not sure if this is needed as disconnect will set the drone instance to null and DroneData will set isConnected to false

File: app/src/main/java/ch/epfl/sdp/drone3d/ui/drone/DroneConnectActivity.kt [hunk 2]
```diff
@@ -7,42 +7,172 @@ package ch.epfl.sdp.drone3d.ui.drone
...
+
     @Inject
     lateinit var droneService: DroneService
 
+    lateinit var droneConnectButton: Button
+    lateinit var simulationConnectButton: Button
+    lateinit var loadingProgressBar: ProgressBar
+    lateinit var ipText: EditText
+    lateinit var portText: EditText
+    lateinit var waitingText: TextView
```
Comment (GabrielFleischer) [inline app/src/main/java/ch/epfl/sdp/drone3d/ui/drone/DroneConnectActivity.kt:49]: Could those fields be private ?

**Review by GabrielFleischer — CHANGES_REQUESTED** [review]

**Review by GabrielFleischer — APPROVED** [review]
All good to me

File: app/src/main/java/ch/epfl/sdp/drone3d/ui/drone/DroneConnectActivity.kt [hunk 3]
```diff
@@ -7,42 +7,172 @@ package ch.epfl.sdp.drone3d.ui.drone
...
+        simulationConnectButton.visibility = View.VISIBLE
+        ipText = findViewById(R.id.text_IP_address)
+        ipText.visibility = View.VISIBLE
+        portText = findViewById(R.id.text_port)
+        portText.visibility = View.VISIBLE
+
+        loadingProgressBar = findViewById(R.id.progress_bar_drone)
+        loadingProgressBar.visibility = View.GONE
+        waitingText = findViewById(R.id.waiting_connection)
+        waitingText.visibility = View.GONE
```
Comment (Jorque) [inline app/src/main/java/ch/epfl/sdp/drone3d/ui/drone/DroneConnectActivity.kt:67]: Maybe use showConnectionOptions() to set visibility

File: app/src/main/java/ch/epfl/sdp/drone3d/ui/drone/DroneConnectActivity.kt [hunk 4]
```diff
@@ -7,42 +7,172 @@ package ch.epfl.sdp.drone3d.ui.drone
...
+                        } else {
+                            droneService.disconnect()
+                            showConnectionOptions()
+                            ToastHandler.showToastAsync(applicationContext, R.string.ip_connection_timeout)
+                        }
+                    }
+
+                    mainHandler.post(myRunnable)
+                }
+            }
```
Comment (Jorque) [inline app/src/main/java/ch/epfl/sdp/drone3d/ui/drone/DroneConnectActivity.kt:99]: This code looks great ! The guy who wrote it must be a genius ! Kappa

**Review by Jorque — APPROVED** [review]
Good code. Could just refractor onCreate of DroneConnectActivity

**Pull request merged**